src/search/search.component.ts

```diff
@@ -0,0 +1,220 @@
+import {
+	Component,
+	Input,
+	EventEmitter,
+	Output
+} from "@angular/core";
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+
+/**
+ * Used to emit changes performed on search components.
+ * @export
+ * @class SearchChange
+ */
+export class SearchChange {
+	/**
+	 * Contains the `Search` that has been changed.
+	 * @type {Search}
+	 * @memberof SearchChange
+	 */
+	source: Search;
+	/**
+	 * The value of the `Search` field encompassed in the `SearchChange` class.
+	 * @type {string}
+	 * @memberof SearchChange
+	 */
+	value: string;
+}
+
+/**
+ * @export
+ * @class Search
+ * @implements {ControlValueAccessor}
+ */
+@Component({
+	selector: "ibm-search",
+	template: `
+		<div class="bx--form-item">
```

Comment:
It looks like this div is just a container div. It's usually possible to remove it and let component tag be the container. In that case class is assigned with a `HostBinding`.

See `src/accordion/accordion-item.component.ts` as an example.